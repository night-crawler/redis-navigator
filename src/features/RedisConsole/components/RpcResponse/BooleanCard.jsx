import React from 'react';
import { Card } from 'semantic-ui-react';
import PropTypes from 'prop-types';


BooleanCard.propTypes = {
    result: PropTypes.bool.isRequired,
};
export default function BooleanCard(props) {
    const { result } = props;
    return <Card fluid={ true } header='Boolean' description={ `${result}` } />;
}
