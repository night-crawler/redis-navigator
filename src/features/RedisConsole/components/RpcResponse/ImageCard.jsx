import PropTypes from 'prop-types';
import React from 'react';
import { Image } from 'semantic-ui-react';


ImageCard.propTypes = {
  dataUri: PropTypes.string.isRequired,
};
export default function ImageCard(props) {
  const { dataUri } = props;
  return <Image className='ImageCard' src={ dataUri } size='medium' />;
}
