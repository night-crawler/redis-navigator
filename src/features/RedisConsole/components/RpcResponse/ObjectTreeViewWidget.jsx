import React from 'react';
import ReactJson from 'react-json-view';
import PropTypes from 'prop-types';


ObjectTreeViewWidget.propTypes = {
    result: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
    groupArraysAfterLength: PropTypes.number,
};
export default function ObjectTreeViewWidget(props) {
    const { result, groupArraysAfterLength = 20 } = props;
    return (
        <ReactJson
            src={ result }
            groupArraysAfterLength={ groupArraysAfterLength }
            name={ false }
        />
    );
}
