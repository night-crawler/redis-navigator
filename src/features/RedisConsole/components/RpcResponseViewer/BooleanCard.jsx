import PropTypes from 'prop-types';
import React from 'react';
import { Card } from 'semantic-ui-react';


BooleanCard.propTypes = {
  result: PropTypes.bool.isRequired,
};
export function BooleanCard(props) {
  const { result } = props;
  return <Card
    className='BooleanCard'
    fluid={ true }
    header='Boolean'
    description={ `${result}` }
  />;
}
