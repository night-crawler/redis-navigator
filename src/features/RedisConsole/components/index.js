import { RedisConsole } from './RedisConsole';

export { RedisConsole };
