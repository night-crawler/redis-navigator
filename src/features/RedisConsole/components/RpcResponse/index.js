import RpcResponse from './Response';

export {
    RpcResponse,
};

export default RpcResponse;
