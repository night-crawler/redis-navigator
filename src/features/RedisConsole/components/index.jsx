import RedisConsole from './RedisConsole';

export default RedisConsole;
