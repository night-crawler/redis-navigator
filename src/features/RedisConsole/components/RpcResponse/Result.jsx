import { isArray, isBoolean, isPlainObject, isString } from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';
import { Card, Header, Segment } from 'semantic-ui-react';

import { TextareaSpoiler } from '~/features/Common/components';

import { isJson, MimeDetector, isBase64, isValidNumber, isYaml } from '~/utils';

import { BooleanCard } from './BooleanCard';
import { ImageCard } from './ImageCard';
import { ObjectCard } from './ObjectCard';
import { ObjectTreeViewWidget } from './ObjectTreeViewWidget';


StringCard.propTypes = {
  result: PropTypes.string.isRequired,
};
export function StringCard(props) {
  const { result } = props;

  let innerResult = false, type = '';

  if (isBase64(result, 4)) {
    innerResult = atob(result);
    type = 'base64';
  } else if (isValidNumber(result)) {
    // just deny json conversion
  }

  if (innerResult) {
    return (
      <Segment basic={ true }>
        <Header as='h5'>String[{ result.length }], { type }</Header>
        <TextareaSpoiler className='left floated' result={ result } />
        { <Result result={ innerResult } /> }
      </Segment>
    );
  }

  return (
    <Card fluid={ true }>
      <Card.Content>
        <Card.Header>String[{ result.length }]{ type ? ', ' : '' }{ type }</Card.Header>
        <Card.Description>
          <pre>{ result }</pre>
        </Card.Description>
      </Card.Content>
    </Card>
  );
}


Result.propTypes = {
  result: PropTypes.any,
};

export function Result(props) {
  const
    { result } = props,
    detector = new MimeDetector(result);

  if (detector.isImage)
    return <ImageCard dataUri={ detector.imageDataURI } />;

  if (isArray(result))
    return <ObjectTreeViewWidget result={ result } />;

  if (isPlainObject(result))
    return <ObjectTreeViewWidget result={ result } />;

  if (isBoolean(result))
    return <BooleanCard result={ result } />;

  if (isJson(result))
    return <ObjectCard result={ result } mode='json' />;

  if (isYaml(result))
    return <ObjectCard result={ result } mode='yaml' />;

  if (isString(result))
    return <StringCard result={ result } />;

  if (result === null)
    return <div>null</div>;

  return <div>{ result }</div>;
}

