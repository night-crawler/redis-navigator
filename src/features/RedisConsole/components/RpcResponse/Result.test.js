/* eslint-disable react/react-in-jsx-scope,no-undef */
import BooleanCard from './BooleanCard';
import ObjectTreeViewWidget from './ObjectTreeViewWidget';
import ImageCard from './ImageCard.jsx';
import Result, { StringCard } from './Result';

import { TextareaSpoiler } from 'features/Common/components';


const GIF = '' +
    'R0lGODlhYABgAIABAAAAAP///yH5BAEKAAEALAAAAABgAGAAAAL+jI+py+0Po5y02ouz3rz7D4biSJbmiabqyrbuC8fyTNcqANguzuf63cP9UMHg' +
    'kFQsHj/Jpm+5cSYdPEMPilBmrwypEBvQJr5d73NpvJjPR67ajHVj4Gi5xglNd6Z5e1Tf5vdX1UfIJKhjeIhYwfjGtgc4J5lByWHZKHbnOKn4iJnp' +
    'GSkaCkphWkmW2jRIWqa6wAlB1wnbgIr6sLbpOmaaS0VbC6mgWSyru9Z7i8sJHKxsm+xMvSwRLQ1tnWt9zRf2vMVoHJttQRnu60p+bD4B6j6L3twt' +
    'Xz9cxTp9z8x/jm3PXzmBpfB4Q6aNGAgvpS4hdJjuAMF28UJEBFdxn0KfExMvSmRXKOMrjy9IYjTZgqRBMFYeqvvGcuJHfTFFjoRZR+auOC7FSQGj' +
    'cmdOmy9/Du2oh6YNnAezofQgrGk/maOEBtwIkgi2jUW1xdhaMWvLGWCvcl0K0CzLmQwjPN2h1G1PtETZ1l1r9i7eqXP33qTq12ffwEUBE245+LDg' +
    's4qvNm7F+HFCyav0Uj4Z+XI5zZw7e/4MOrTo0aRLnygAADs=';

const GIF_BIN = atob(GIF);


describe('Result', () => {
    it('should render Result.StringCard', () => {
        const wrapper = shallow(<Result result='1' />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.containsMatchingElement(<StringCard result='1' />)).toEqual(true);
    });

    it('should render Result.BooleanCard', () => {
        const wrapper = shallow(<Result result={ true } />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.containsMatchingElement(<BooleanCard result={ true } />)).toEqual(true);
    });

    it('should render Result.ObjectTreeViewWidget[Array]', () => {
        const wrapper = shallow(<Result result={ [1, 2, 3] } />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<ObjectTreeViewWidget result={ [1, 2, 3] } />)).toEqual(true);
    });

    it('should render Result.ObjectTreeViewWidget[Object]', () => {
        const wrapper = shallow(<Result result={ { a: 1, b: 2, c: 3 } } />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<ObjectTreeViewWidget result={ { a: 1, b: 2, c: 3 } } />)).toEqual(true);
    });

    it('should render null', () => {
        const wrapper = shallow(<Result result={ null } />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<div>null</div>)).toEqual(true);
    });

    it('should render image', () => {
        const wrapper = shallow(<Result result={ GIF_BIN } />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(ImageCard)).toHaveLength(1);
    });
});


describe('StringCard', () => {
    it('should render TextareaSpoiler for base64', () => {
        const wrapper = shallow(<StringCard result='buckbuck' />);
        expect(wrapper.find(TextareaSpoiler)).toHaveLength(1);
    });

    it('should not treat strings shorter 4 chars as b64', () => {
        const wrapper = shallow(<StringCard result='buck' />);
        expect(wrapper.find(TextareaSpoiler)).toHaveLength(0);
    });
});