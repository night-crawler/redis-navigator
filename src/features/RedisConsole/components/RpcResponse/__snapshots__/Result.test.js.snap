// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should render Result.BooleanCard 1`] = `
<BooleanCard
  result={true}
/>
`;

exports[`Result should render Result.ObjectTreeViewWidget[Array] 1`] = `
<ObjectTreeViewWidget
  result={
    Array [
      1,
      2,
      3,
    ]
  }
/>
`;

exports[`Result should render Result.ObjectTreeViewWidget[Object] 1`] = `
<ObjectTreeViewWidget
  result={
    Object {
      "a": 1,
      "b": 2,
      "c": 3,
    }
  }
/>
`;

exports[`Result should render Result.StringCard 1`] = `
<StringCard
  result="1"
/>
`;

exports[`Result should render image 1`] = `
<ImageCard
  dataUri="data:image/gif;charset=utf-8;base64,R0lGODlhYABgAIABAAAAAP///yH5BAEKAAEALAAAAABgAGAAAAL+jI+py+0Po5y02ouz3rz7D4biSJbmiabqyrbuC8fyTNcqANguzuf63cP9UMHgkFQsHj/Jpm+5cSYdPEMPilBmrwypEBvQJr5d73NpvJjPR67ajHVj4Gi5xglNd6Z5e1Tf5vdX1UfIJKhjeIhYwfjGtgc4J5lByWHZKHbnOKn4iJnpGSkaCkphWkmW2jRIWqa6wAlB1wnbgIr6sLbpOmaaS0VbC6mgWSyru9Z7i8sJHKxsm+xMvSwRLQ1tnWt9zRf2vMVoHJttQRnu60p+bD4B6j6L3twtXz9cxTp9z8x/jm3PXzmBpfB4Q6aNGAgvpS4hdJjuAMF28UJEBFdxn0KfExMvSmRXKOMrjy9IYjTZgqRBMFYeqvvGcuJHfTFFjoRZR+auOC7FSQGjcmdOmy9/Du2oh6YNnAezofQgrGk/maOEBtwIkgi2jUW1xdhaMWvLGWCvcl0K0CzLmQwjPN2h1G1PtETZ1l1r9i7eqXP33qTq12ffwEUBE245+LDgs4qvNm7F+HFCyav0Uj4Z+XI5zZw7e/4MOrTo0aRLnygAADs="
/>
`;

exports[`Result should render null 1`] = `
<div>
  null
</div>
`;
